sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs

```diff
@@ -369,6 +382,249 @@ async Task<Result> RuntimeGetProperties (MessageId id, DotnetObjectId objectId,
+			Log ("info", $"\tmethod {method_name} location: {location}");
+			frame = new Frame (method, location, frame_id - 1);
+
+			return new {
+				functionName = method_name,
+				callFrameId = $"dotnet:scope:{frame_id - 1}",
+				functionLocation = method.StartLocation.AsLocation (),
+
+				location = location.AsLocation (),
+
+				url = store.ToUrl (location),
+
+				scopeChain = new [] {
+					new {
+						type = "local",
+						@object = new {
+							@type = "object",
+							className = "Object",
+							description = "Object",
+							objectId = $"dotnet:scope:{frame_id-1}",
+						},
+						name = method_name,
+						startLocation = method.StartLocation.AsLocation (),
+						endLocation = method.EndLocation.AsLocation (),
+					}}
+			};
+		}
+
+		void GetMonoFrames (ExecutionContext context, DebugStore store, List<object> callFrames, IEnumerable<JObject> the_mono_frames)
+		{
+			var frames = new List<Frame> ();
+			int frame_id = 0;
+
+			foreach (var mono_frame in the_mono_frames) {
+				var call_frame = GetMonoFrame (store, mono_frame, ++frame_id, out var frame);
+				frames.Add (frame);
+
+				callFrames.Add (call_frame);
+			}
+
+			context.CallStack = frames;
+		}
+
+		async Task<IDictionary<string, JObject>> GetJavaScriptVariables (SessionId id, JObject frame, CancellationToken token)
+		{
+			// We are stopped on a managed -> javascript call - like for instance mono_wasm_fire_exc().
+			Log ("verbose", $"getting javascript variables");
+			var scopeChain = frame ["scopeChain"]?.Values<JObject> ();
+			if (scopeChain == null || scopeChain.Count () == 0) {
+				Log ("info", $"Frame is missing scopeChain");
+				return null;
+			}
+
+			var scopeObj = scopeChain.First () ["object"]?.Value<JObject> ();
+			if (scopeObj == null) {
+				Log ("info", $"Frame is missing scope object");
+				return null;
+			}
+
+			var objectId = scopeObj ["objectId"]?.Value<string> ();
+			if (objectId == null) {
+				Log ("info", $"missing objectId");
+				return null;
+			}
+
+			var cmd_args = JObject.FromObject (new {
+				objectId,
+				ownProperties = false,
+				accessorPropertiesOnly = false,
+				generatePreview = true
+			});
+
+			var cmd_res = await SendCommand (id, "Runtime.getProperties", cmd_args, token);
+			if (!cmd_res.IsOk) {
+				Log ("info", $"command failed");
+				return null;
+			}
+
+			Log ("verbose", $"got properties: {cmd_res}");
+			var variables = cmd_res.Value ["result"]?.Values<JObject> ();
+			if (variables == null || variables.Count () == 0) {
+				Log ("info", $"did not get any variables");
+				return null;
+			}
+
+			var dict = new Dictionary<string, JObject> ();
+
+			foreach (var variable in variables) {
+				Log ("verbose", $"got variable: {variable}");
+				var name = variable ["name"]?.Value<string> ();
+				if (name != null)
+					dict.Add (name, variable);
+			}
+
+			return dict;
+		}
+
+		string GetVariableValue (JObject variable, string expectedType)
```

Comment:
I don't like the idea of using a generic method here too much because we only care about two types: "string" and "number" (which could be an int or long).